src/event.rs

```diff
@@ -590,7 +548,9 @@ pub enum WindowEvent<'a> {
     /// For more information about DPI in general, see the [`dpi`](crate::dpi) module.
     ScaleFactorChanged {
         scale_factor: f64,
-        new_inner_size: &'a mut PhysicalSize<u32>,
+        /// The pointer could be upgraded successfully only from the callback for this event,
+        /// otherwise it'll always fail to do so.
+        new_inner_size: Weak<Mutex<PhysicalSize<u32>>>,
```

Comment:
Might make sense to wrap this in a struct, to hide the specific `Weak` and `Mutex` implementation that we use?